### ISSUE TITLE
pm.find_MAP breaks with progressbar=False

### ISSUE THREAD

**Issue by catethos** [pymc-devs/pymc#2642]
Hi,
There is a problem when using `pm.find_MAP` without the progress bar.
```
with pm.Model():
    n = pm.Normal('normal',mu=0,sd=1)
    ML = pm.find_MAP(progressbar=False)
```

```
AttributeError                            Traceback (most recent call last)
<ipython-input-25-f3fd7255be59> in <module>()
      1 with pm.Model():
      2     n = pm.Normal('normal',mu=0,sd=1)
----> 3     ML = pm.find_MAP(progressbar=False)

/Users/catethos/anaconda3/lib/python3.6/site-packages/pymc3/tuning/starting.py in find_MAP(start, vars, method, return_raw, include_transformed, progressbar, maxeval, model, *args, **kwargs)
    121             opt_result = minimize(cost_func, x0, method=method, jac=compute_gradient, *args, **kwargs)
    122             mx0 = opt_result["x"]  # r -> opt_result
--> 123             cost_func.progress.total = cost_func.progress.n + 1
    124             cost_func.progress.update()
    125         except (KeyboardInterrupt, StopIteration) as e:

AttributeError: 'tqdm' object has no attribute 'n'
```
I have asked the same problem [here](https://discourse.pymc.io/t/pm-find-map-break-with-progressbar-false/449)

## Versions and main components

* PyMC3 Version: 3.2
* Theano Version: 0.9.0
* Python Version: 3.6
* Operating system: OSX
* How did you install PyMC3: conda

**Comment by kyleabeauchamp** [comment]
This looks like a known a tqdm thread safety issue.  Are you running
multiple threads?

On Oct 19, 2017 4:21 AM, "Caleb Foong" <notifications@github.com> wrote:

Hi,
There is a problem when using pm.find_MAP without the progress bar.

with pm.Model():
    n = pm.Normal('normal',mu=0,sd=1)
    ML = pm.find_MAP(progressbar=False)

AttributeError                            Traceback (most recent call last)
<ipython-input-25-f3fd7255be59> in <module>()
      1 with pm.Model():
      2     n = pm.Normal('normal',mu=0,sd=1)
----> 3     ML = pm.find_MAP(progressbar=False)

/Users/catethos/anaconda3/lib/python3.6/site-packages/pymc3/tuning/starting.py
in find_MAP(start, vars, method, return_raw, include_transformed,
progressbar, maxeval, model, *args, **kwargs)
    121             opt_result = minimize(cost_func, x0,
method=method, jac=compute_gradient, *args, **kwargs)
    122             mx0 = opt_result["x"]  # r -> opt_result
--> 123             cost_func.progress.total = cost_func.progress.n + 1
    124             cost_func.progress.update()
    125         except (KeyboardInterrupt, StopIteration) as e:

AttributeError: 'tqdm' object has no attribute 'n'

I have asked the same problem here
<https://discourse.pymc.io/t/pm-find-map-break-with-progressbar-false/449>
Versions and main components

   - PyMC3 Version: 3.2
   - Theano Version: 0.9.0
   - Python Version: 3.6
   - Operating system: OSX
   - How did you install PyMC3: conda

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/pymc-devs/pymc3/issues/2642>, or mute the thread

**Comment by catethos** [comment]
Hi @kyleabeauchamp, no I am not using multiple threads.

**Comment by twiecki** [comment]
This looks like a proper bug where we try to access the tqdm even if `progressbar==False`. Should be easy to fix

**Comment by ferrine** [comment]
`tqdm.trange(n, disable=not progressbar)` should fix it

**Comment by ferrine** [comment]
interestingly we have that line in master https://github.com/pymc-devs/pymc3/blob/master/pymc3/tuning/starting.py#L170

**Comment by twiecki** [comment]
Fixed by #2649.
